frontend/packages/console-dynamic-plugin-sdk/src/api/api-types.ts

```diff
@@ -43,3 +43,15 @@ export type UseK8sWatchResources = <R extends ResourcesObject>(
 export type UseResolvedExtensions = <E extends Extension>(
   ...typeGuards: ExtensionTypeGuard<E>[]
 ) => [ResolvedExtension<E>[], boolean, any[]];
+
+export type CoFetch = (url: string, options?: RequestInit, timeout?: number) => Promise<Response>;
+
+export type CoFetchJSON<T = any> = {
+  (url: string, method?: string, options?: RequestInit, timeout?: number): Promise<T>;
+  delete(url: string, options?: RequestInit, json?: any, timeout?: number): Promise<T>;
```

Comment:
I realize that this is what the existing API is, but I think it would be better to keep the args in the same order (json before options) even if json is less common for delete requests.

Reply:
Done. Updated PR.